src/DocumentDb/DocumentDbDocumentAdapter.cs

```diff
@@ -395,6 +395,7 @@ public partial class DocumentDbAdapter : IDocumentDbAdapter
 		/// <returns>Replaced document.</returns>
 		public async Task<ResourceResponse<Document>> ReplaceDocumentAsync<T>(
 			string dbId, string collectionId, string docId, T entity, RequestOptions requestOptions = null)
+            where T:class
```

Comment:
Please use tabs here and in other places

Reply:
Fixed, thanks.

Reply:
Thanks, it looks like we lost .editorconfing during the extraction of this repo. I'll create PR to add one.

Reply:
#133

Reply:
There is no such constraint on the underlying DocumentDb client SDK and so the wrapper should try to avoid introducing the constraints unless it is required for the purpose of the method. In this case and in many methods in this api, T can be a struct type as well as a class and this should still work.

---
In reply to: [335635225](https://github.com/microsoft/Omex/pull/132#discussion_r335635225) [](ancestors = 335635225)

Reply:
I've removed constraints and created a separate method for the null check with type validation, thanks. Could you please check again?